File: LoanCalculator/Models/LoanParameters.swift
import Foundation

struct LoanParameters: Equatable {
    var loanAmount: Double
    var interestRate: Double
    var termYears: Double
    var paymentsPerYear: PaymentFrequency

    init(
        loanAmount: Double = 350_000,
        interestRate: Double = 5.5,
        termYears: Double = 30,
        paymentsPerYear: PaymentFrequency = .monthly
    ) {
        self.loanAmount = loanAmount
        self.interestRate = interestRate
        self.termYears = termYears
        self.paymentsPerYear = paymentsPerYear
    }
}

enum PaymentFrequency: Int, CaseIterable, Identifiable, Codable {
    case monthly = 12
    case biWeekly = 26
    case weekly = 52

    var id: Int { rawValue }

    var label: String {
        switch self {
        case .monthly: return "Monthly"
        case .biWeekly: return "Bi-Weekly"
        case .weekly: return "Weekly"
        }
    }
}

File: src/models/LoanParameters.ts
export enum PaymentFrequency {
  MONTHLY = 12,
  BI_WEEKLY = 26,
  WEEKLY = 52,
}

export interface LoanParameters {
  loanAmount: number;
  interestRate: number;
  termYears: number;
  paymentsPerYear: PaymentFrequency;
}

export const createDefaultLoanParameters = (
  loanAmount: number = 350_000,
  interestRate: number = 5.5,
  termYears: number = 30,
  paymentsPerYear: PaymentFrequency = PaymentFrequency.MONTHLY
): LoanParameters => ({
  loanAmount,
  interestRate,
  termYears,
  paymentsPerYear,
});

export const getPaymentFrequencyLabel = (frequency: PaymentFrequency): string => {
  switch (frequency) {
    case PaymentFrequency.MONTHLY:
      return 'Monthly';
    case PaymentFrequency.BI_WEEKLY:
      return 'Bi-Weekly';
    case PaymentFrequency.WEEKLY:
      return 'Weekly';
    default:
      return 'Monthly';
  }
};

export const getAllPaymentFrequencies = () => [
  PaymentFrequency.MONTHLY,
  PaymentFrequency.BI_WEEKLY,
  PaymentFrequency.WEEKLY,
];